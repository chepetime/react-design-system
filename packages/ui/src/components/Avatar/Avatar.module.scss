@use "./../../styles" as styles;

.Avatar {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  display: inline-flex;
  justify-content: center;
  align-items: center;
  font-size: 1rem;
  font-weight: bold;
  border-radius: 50%;
  color: #fff;
  background-color: #ccc;
  overflow: hidden;
}

.small {
  width: 2.5rem;
  height: 2.5rem;
}

.medium {
  width: 3.75rem;
  height: 3.75rem;
}

.large {
  width: 5rem;
  height: 5rem;
}
