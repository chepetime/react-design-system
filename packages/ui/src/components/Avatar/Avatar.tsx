import * as React from "react";
import classnames from "classnames";
import css from "./Avatar.module.scss";

export interface AvatarProps {
  children?: React.ReactNode;
}

export function Avatar(props: AvatarProps) {
  return (
    <div {...props} className={classnames(css.Avatar)}>
      <p>Avatar component working!</p>
      <div>{props.children}</div>
    </div>
  );
}

Avatar.displayName = "Avatar";
