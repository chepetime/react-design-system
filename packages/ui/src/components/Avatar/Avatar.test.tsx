import React from "react";
import { render } from "@testing-library/react";
import { Avatar } from "./Avatar";

describe("Avatar", () => {
  test("renders the Avatar component", () => {
    render(<Avatar />);
  });
});
