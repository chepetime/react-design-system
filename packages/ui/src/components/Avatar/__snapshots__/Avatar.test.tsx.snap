// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<div>
  <div
    class="Avatar"
  >
    <p>
      Avatar component working!
    </p>
    <div />
  </div>
</div>
`;
